### PR TITLE
PCHR-2002: Add css for absence tab page

### PR DESCRIPTION
**Pull request by dhavalsharma**
Added the styles for the absence tab page.

## Screenshot

image

### CODE REVIEW

**Comment by deb1990** [comment]
1. "Create New" should be in block letters.
2. Record staff sickness and Record staff overtime dropdown not working.
3. Tabs dont have hand cursor

**Comment by deb1990** [comment]
civihr_staff_compucorp_co_uk___hr17
Arrow not center alighned

**Comment by deb1990** [comment]
civihr_staff_compucorp_co_uk___hr17
As hover style is not mentioned in the mock, lets follow what we have. That is, there should not be any style change on hovered. Please refer My Leave or Manager Leave page.

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 1]
```diff
@@ -0,0 +1,47 @@
+.chr_leave-absences-admin {
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: Is this required? Cant find it being used in the code.

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  background: transparent;
+  box-shadow: none;
+  padding: 0;
+}
+
+.chr_absence-tab-page {
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: `absence-tab-page`

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  .button-container {
+    margin-bottom: 20px;
+  }
+
+  .tabs-container {
+
+    ul {
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: Not required when you put `<div class="container">` in the html

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      font-weight: 200;
+    }
+
+    > .btn-primary {
+      min-width: 0px;
+    }
+  }
+}
+
+.chr_absence-tab-page--active_bottom {
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: `chr_absence-tab-page--active_bottom` should not be created. Active class is already present in tabs, see how it is done for My Leave and Manager leave(Blue bottom border). And just override it.

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        }
+
+        a {
+          font-size: 14px;
+        }
+      }
+    }
+  }
+
+  .btn-group {
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss:50]: Do we need to override these settings?

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss (hunk 2 above)
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: A lot of things are changed for the tabs which seems unnecessary to me as the tabs are same as manager and my leave. Only the active tab color and `button-container` margin seems different to me.

**Review by deb1990 — CHANGES_REQUESTED** [review]

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss (hunk 5 above)
Comment (dhavalsharma) [reply civihr_default_theme/scss/pages/_admin_absence_tab.scss:50]: yes to reduce the width and added comment for rest

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss (hunk 3 above)
Comment (dhavalsharma) [reply civihr_default_theme/scss/pages/_admin_absence_tab.scss]: ul is required to customize certain styles to match design like border bottom etc.,

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss (hunk 2 above)
Comment (dhavalsharma) [reply civihr_default_theme/scss/pages/_admin_absence_tab.scss]: the design is totally different from what is there in my leave or manager leave, see the comments in css

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 6]
```diff
@@ -1,47 +1,68 @@
...
-
-.chr_absence-tab-page {
-
+.absence-tab-page {
   .button-container {
     margin-bottom: 20px;
   }
 
   .tabs-container {
+    .container {
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: Line break

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 7]
```diff
@@ -1,47 +1,68 @@
...
 
       li {
-
         &:first-child {
+          /*to match the shift from left in the design*/
           margin-left: 20px;
         }
 
         a {
           font-size: 14px;
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: Use SCSS variables, you may need to create new variable for 14px

File: civihr_default_theme/scss/base/_variables.scss [hunk 8]
```diff
@@ -29,6 +29,7 @@ $crm-grizzly:         #dcdddd !default;
 $crm-color-neutral:   #949caa !default; // gray shade
 $crm-grayish-blue:    #f4f7f8 !default;
 
+$crm-dark:               #000000 !default;
```
Comment (deb1990) [inline civihr_default_theme/scss/base/_variables.scss]: Indentation does not match, also in the design the color is not black its, '4c4c5d'. Please use $gray-darker:

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 9]
```diff
@@ -1,47 +1,68 @@
...
 
   .tabs-container {
+    .container {
+      /*to shift the container to left else it was not aligned with button*/
+      padding-left: 0;
+    }
 
     ul {
-      margin-left: 0px;
+      margin-left: 0;
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: `margin: 0;` would be better as we dont want any margin.

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 10]
```diff
@@ -1,47 +1,68 @@
...
+      /**/
       background: white;
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: Use variables, no color should be directly used

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 11]
```diff
@@ -1,47 +1,68 @@
...
         a {
           font-size: 14px;
+          cursor: pointer;
+          /*otherwise it jumps in on hover*/
+          border-width: 0px;
+
+          &:hover {
+            /*as by default it sets it to #fbfbfc*/
+            background-color: transparent;
+            border-width: 0px;
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: I removed it and did not see any difference.

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 12]
```diff
@@ -1,47 +1,68 @@
...
+          }
+        }
+
+        &.active {
+          a {
+            &,
+            &:focus,
+            &:hover {
+              border: 0 !important;
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: `!imporant` should be used only if necessary. Here I removed and it still works.

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 13]
```diff
@@ -1,47 +1,68 @@
...
       }
     }
   }
 
   .btn-group {
-
     > .btn {
       border-color: transparent;
+      /*reduce the font weight to meet design*/
       font-weight: 200;
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: Use font-weight variables

**Review by deb1990 — CHANGES_REQUESTED** [review]
Please update screenshot, also provide screenshot when the dropdown is open.

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss (hunk 9 above)
Comment (dhavalsharma) [reply civihr_default_theme/scss/pages/_admin_absence_tab.scss]: rest of the margins are already 0 , so not re-setting again

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss (hunk 11 above)
Comment (dhavalsharma) [reply civihr_default_theme/scss/pages/_admin_absence_tab.scss]: if i remove this is what i get 
image

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+      li {
+        &:first-child {
+          /*to match the shift from left in the design*/
+          margin-left: 20px;
+        }
+
+        a {
+          font-size: $crm-font-size-medium;
+          /*otherwise it jumps in on hover*/
+          border-width: 0px;
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: Dont use unit with 0.

File: civihr_default_theme/scss/pages/_admin_absence_tab.scss [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+
+          &:hover {
+            /*as by default it sets it to #fbfbfc*/
+            background-color: transparent;
+            border-width: 0px;
```
Comment (deb1990) [inline civihr_default_theme/scss/pages/_admin_absence_tab.scss]: Dont use unit with 0.

**Review by deb1990 — CHANGES_REQUESTED** [review]

**Comment by davialexandre** [comment]
I'm closing this as it became stale and we probably don't need it anymore. I'll keep the branch for now. If we really don't need it, the branch will be deleted on the next branch cleanup

**Pull request closed without merging**